enterprise/api/filters.py

```diff
@@ -54,3 +54,24 @@ def filter_queryset(self, request, queryset, view):
             queryset = queryset.filter(user_id=request.user.id)
 
         return queryset
+
+
+class EnterpriseCustomerFilterBackend(filters.BaseFilterBackend):
+    """
+    Filter backend for `EnterpriseCustomerViewSet`
+    """
+
+    def filter_queryset(self, request, queryset, view):
+        """
+        Filter out enterprise customer if learner is not linked
+        """
+        enterprise_slug = request.GET.get('slug')
```

Comment:
@adamstankiewicz I made this change intentionally. This endpoint is used at many places and to be on the safe side and to avoid issues I just want to do filtering on the bases of `slug` only.

Reply:
@muhammad-ammar I understand that rationale, but we're only fixing the issue partially if we do the filtering only when the `slug` query parameter is supplied. An unlinked learner will still have access to their previously linked EnterpriseCustomer's data through the endpoint, which is still the underlying issue here.

That said, I think it's worth fixing more broadly than only when `slug` is provided.